src/app/services/application.service.ts

```diff
@@ -301,99 +269,148 @@ export class ApplicationService {
   }
 
   /**
-   * Returns status abbreviation.
+   * Map Tantalis Status to status code.
    */
-  getStatusCode(status: string): string {
-    if (status) {
-      switch (status.toUpperCase()) {
-        case 'ABANDONED': return this.ABANDONED;
-        case 'ACCEPTED': return this.ACCEPTED;
-        case 'ALLOWED': return this.ALLOWED;
-        case 'CANCELLED': return this.CANCELLED;
-        case 'DISALLOWED': return this.DISALLOWED;
-        case 'DISPOSITION IN GOOD STANDING': return this.DISPOSITION_GOOD_STANDING;
-        case 'OFFER ACCEPTED': return this.OFFER_ACCEPTED;
-        case 'OFFER NOT ACCEPTED': return this.OFFER_NOT_ACCEPTED;
-        case 'OFFERED': return this.OFFERED;
-        case 'SUSPENDED': return this.SUSPENDED;
+  getStatusCode(statusString: string): string {
+    if (statusString) {
+      switch (statusString.toUpperCase()) {
+        case 'ABANDONED':
+        case 'CANCELLED':
+        case 'OFFER NOT ACCEPTED':
+        case 'OFFER RESCINDED':
+        case 'RETURNED':
+        case 'REVERTED':
+        case 'SOLD':
+        case 'SUSPENDED':
+        case 'WITHDRAWN':
+          return this.ABANDONED;
+
+        case 'ACCEPTED':
+        case 'ALLOWED':
+        case 'PENDING':
+        case 'RECEIVED':
+          return this.APPLICATION_UNDER_REVIEW;
+
+        case 'OFFER ACCEPTED':
+        case 'OFFERED':
+          return this.APPLICATION_REVIEW_COMPLETE;
+
+        case 'ACTIVE':
+        case 'COMPLETED':
+        case 'DISPOSITION IN GOOD STANDING':
+        case 'EXPIRED':
```

Comment:
Is an expired application really approved? Or is this the retired status, where it was completed, but not is too old?

Also, is there some place where the meaning of these statuses is listed? Or is that a question for the business/POs?

Reply:
The quick answer is yes, Expired counts as Decision Approved.
I can't speak to the business directly but Lena did and documented everything in PRC-859 (esp the Tantalis Status Mapping doc).